### PR TITLE
refactor: delete old code

### LINKED ISSUE #14738: ng.probe is not available even in development mode
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

ng.probe is missing when I open up the console.

**Expected behavior**

ng.probe should be available.

**Minimal reproduction of the problem with instructions**

Just use the latest version of Angular, 4.0.0-rc.1, at the time of this report.

1. Open up the console.
2. Change the console context to plunkerPreviewTarget
2. Type ng.probe and evaluate in the REPL.

Demos:
Working in a previous version of Angular: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5?p=preview (screenshot: http://i.imgur.com/eYw1QpP.png)
Broken in the newest version of Angular: http://plnkr.co/edit/YbTVpkpqRTABFw2UxPxI?p=preview (screenshot: http://i.imgur.com/lHKB193.png)

**What is the motivation / use case for changing the behavior?**

I really like the ng.probe debug tool.

**Please tell us about your environment:**

Chrome 56

* **Angular version:** 4.0.0-rc.1

* **Browser:** Chrome 56

 
* **Language:** TypeScript 2.2

* **Node (for AoT issues):** `node --version` =   N/A

### PR DESCRIPTION
**Pull request by tbosch**
fixes #14738

### CODE REVIEW

File: modules/@angular/core/src/core.ts [hunk 1]
```diff
@@ -49,3 +46,15 @@ export type AnimationEntryMetadata = any;
  * @deprecated
  */
 export type AnimationStateTransitionMetadata = any;
+/**
+ * @deprecated
```
Comment (vicb) [inline modules/@angular/core/src/core.ts]: could you add when ? `@deprecated from v4`

File: modules/@angular/core/test/error_handler_spec.ts [hunk 2]
```diff
@@ -99,3 +97,16 @@ Context`);
...
   });
 }
+
+/**
+ * Thrown when an exception was raised during view creation, change detection or destruction.
+ *
+ * This error wraps the original exception to attach additional contextual information that can
+ * be useful for debugging.
+ */
+export function viewWrappedError(originalError: any, context: any): Error {
```
Comment (vicb) [inline modules/@angular/core/test/error_handler_spec.ts]: used only in tests ? 
Prefix with `test` not to confuse it when looking for a symbol ?

**Comment by tbosch** [comment]
Fixes https://github.com/angular/angular/issues/14738 as well.

File: modules/@angular/core/test/error_handler_spec.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/core/test/error_handler_spec.ts]: And I should not export it...

**Review by vicb — APPROVED** [review]
a few nits to update. 
nice to see so many LOC removed !

File: modules/@angular/core/src/core.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/src/core.ts]: Done

File: modules/@angular/core/test/error_handler_spec.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/core/test/error_handler_spec.ts]: Done

**Review by IgorMinar — APPROVED** [review]

**Pull request merged**